### DISCUSSION TITLE
Renovate not showing/failing on unmet peer warnings for pnpm?

### DISCUSSION THREAD

**Discussion by GKersten** [renovatebot/renovate#30304] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure Devops Pipeline

### Please tell us more about your question or problem

We have just setup Renovate and so far seems to work really nice for many different repositories for both Nuget and NPM, now we are also setting up a new monorepo with PNPM which does also seem to be picked up nicely. Renovate has been creating PRs on which we have build validation, so far everything looked good.

Now, we ran into an issue where Renovate created a PR to update:
`"eslint": "^8.0.0"` to `"eslint": "^9.0.0"`

in that same project however, we also have a dependency on: 
`"eslint-plugin-unused-imports": "^3.0.0"` which has a "peerDependency" still on "eslint": "8".
(for which renovate just created a separate PR to update to `"eslint-plugin-unused-imports": "^4.0.0"`, which now has support for "eslint":"9")

We knew there was a conflict here, so that is why it caught our attention, I was expecting Renovate to run into errors when upgrading `eslint` to `^9.0.0`. But, the PR is created with a change in both `package.json` & `pnpm-lock.yaml` to just update "eslint".

The generated lock file did not cause any issues when running `pnpm install`, so also the build pipeline is not failing at this point. 
However when I would run `pnpm install --resolution-only` on the created branch, I am getting the warnings I was expecting:

````
 WARN  Issues with peer dependencies found
packages/eslint-config
├─┬ @typescript-eslint/eslint-plugin 7.14.1
│ ├── ✕ unmet peer eslint@^8.56.0: found 9.7.0
│ └─┬ @typescript-eslint/type-utils 7.14.1
│   ├── ✕ unmet peer eslint@^8.56.0: found 9.7.0
│   └─┬ @typescript-eslint/utils 7.14.1
│     └── ✕ unmet peer eslint@^8.56.0: found 9.7.0
├─┬ @typescript-eslint/parser 7.14.1
│ └── ✕ unmet peer eslint@^8.56.0: found 9.7.0
├─┬ eslint-plugin-import 2.29.1
│ └── ✕ unmet peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8": found 9.7.0
├─┬ eslint-plugin-jest 28.6.0
│ └─┬ @typescript-eslint/utils 7.8.0
│   └── ✕ unmet peer eslint@^8.56.0: found 9.7.0
├─┬ eslint-plugin-rxjs 5.0.3
│ ├── ✕ unmet peer eslint@^8.0.0: found 9.7.0
│ ├─┬ @typescript-eslint/experimental-utils 5.62.0
│ │ ├── ✕ unmet peer eslint@"^6.0.0 || ^7.0.0 || ^8.0.0": found 9.7.0
│ │ └─┬ @typescript-eslint/utils 5.62.0
│ │   └── ✕ unmet peer eslint@"^6.0.0 || ^7.0.0 || ^8.0.0": found 9.7.0
│ └─┬ eslint-etc 5.2.1
│   └── ✕ unmet peer eslint@^8.0.0: found 9.7.0
└─┬ eslint-plugin-unused-imports 3.2.0
  └── ✕ unmet peer eslint@8: found 9.7.0
````

One more thing I tested out is see what happens if I try to perform the update myself from our `main` branch. so I:
- Checked out a clean state for `main`
- `pnpm update eslint@9`
- See the same WARNs as above

Now I start to doubt / have questions:
- Is this expected?
- What steps / commands does renovate exactly take to update a PNPM package? Is it supposed to fail on unmet peerDependencies? Can we configure it like that?
- Is this a configuration issue with Renovate? 
- Is it a misunderstanding of how PNPM is supposed to behave here.

I hope to find some support here in how this is handled for others dealing with package updates with Renovate & pnpm.

If more details are needed please let me know.

**Comment by viceice** [comment]
https://pnpm.io/npmrc#strict-peer-dependencies

you need to configure it manually, not a renovate issue

**Comment by rarkins** [comment, marked as answer]
> Now I start to doubt / have questions:
> 
> * Is this expected?

Yes. If pnpm doesn't error then Renovate won't error

> * What steps / commands does renovate exactly take to update a PNPM package? Is it supposed to fail on unmet peerDependencies? Can we configure it like that?

You need to look up the pnpm docs for this

> * Is this a configuration issue with Renovate?

No

**Comment by GKersten** [comment]
Thank you guys, that was indeed what I needed, now it works. I was sure we enabled it, but apparently we did set it to false explicitly at some point, which was a mistake. Thanks for the quick replies!
